j-lawyer-client/src/com/jdimension/jlawyer/client/utils/FindRouteToAddress.java

```diff
@@ -0,0 +1,777 @@
+
+  You should also get your employer (if you work as a programmer) or school,
+if any, to sign a "copyright disclaimer" for the program, if necessary.
+For more information on this, and how to apply and follow the GNU AGPL, see
+<https://www.gnu.org/licenses/>.
+ */
+package com.jdimension.jlawyer.client.utils;
+
+import com.jdimension.jlawyer.client.editors.EditorsRegistry;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonReader;
+import javax.json.JsonObject;
+import javax.swing.JOptionPane;
+
+/**
+ *
+ * @author Maximilian Steinert
+ */
+public class FindRouteToAddress {
+    
+    public static String getCoordinates(String address) {
+    
+        String encodedAddress = "";
+
+        try {
+            encodedAddress = replaceGermanCharacters(address);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        try {
+            String urlStr = "https://nominatim.openstreetmap.org/search?q="
+                    + encodedAddress.replace("%2B", "+") +"&format=json";
+
+            // JOptionPane.showMessageDialog(null, urlStr);
+            URL url = new URL(urlStr);
+            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+            conn.setRequestMethod("GET");
+            conn.setRequestProperty("User-Agent", "Mozilla/5.0");
+
+            BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()));
+            JsonReader jsonReader = Json.createReader(in);
+            JsonArray jsonArray = jsonReader.readArray();
+            jsonReader.close();
+            in.close();
+
+            if (!jsonArray.isEmpty()) {
+                JsonObject jsonObject = jsonArray.getJsonObject(0);
+                String lat = jsonObject.getString("lat").replace(",", ".");
+                String lon = jsonObject.getString("lon").replace(",", ".");
+                String stringCoordinates = lat + "-"+ lon;
+                return stringCoordinates.replace("-", ",");
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public static void openBrowserWithRoute(String startCoordinates, String endCoordinates) {
+        try {
+            
+            String url = "https://www.openstreetmap.org/directions?engine=fossgis_osrm_car&route="+
+                startCoordinates + ";"+ endCoordinates;
+            
+            try {
+                DesktopUtils.openBrowser(url);
+            } catch (Exception ex) {
+                JOptionPane.showMessageDialog(EditorsRegistry.getInstance().getMainWindow(), "Service-Website kann nicht geöffnet werden: " + ex.getMessage(), "Browserfehler", JOptionPane.ERROR_MESSAGE);
+            }
+            
+        } catch (Exception e) {
+            JOptionPane.showMessageDialog(null, "Fehler: " + e);
```

Comment:
as this is a utility class, it is better to throw / forward exceptions and let the caller decide on how to handle them.